### PR TITLE
Add Blog: OpenSearch Simplified: The Power of Application-Based Configuration Templates 

### PR DESCRIPTION
**Pull request by mgodwan**
### Description
 [BLOG] OpenSearch Simplified: The Power of Application-Based Configuration Templates 
 
### Issues Resolved
https://github.com/opensearch-project/project-website/issues/3508

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# OpenSearch Simplified: The Power of Application-Based Configuration Templates
+
+
+OpenSearch supports a wide variety of use-cases, ranging from logs, metrics, traces, website search, vectors, etc., and enables users to build solutions for various applications based on their use cases. As the use cases for OpenSearch continue to grow, managing indices and configuring them with the right settings can become a daunting task, both for experienced and new users. OpenSearch provides numerous knobs and settings to fine-tune indices for different performance and usability dimensions, such as throughput, latency, disk utilization, and more. However, for new users, finding the optimal configuration often requires extensive experimentation and developer effort, creating friction during the on-boarding process. As new features are developed and released, experienced users of OpenSearch may miss out on them as well.
```
Comment (pajuric) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:4]: Consider adding links to the following: 
- Logs: https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/logs/
- Metrics: https://opensearch.org/docs/latest/monitoring-your-cluster/metrics/getting-started/
- Search: https://opensearch.org/docs/latest/search-plugins/
- Vector: https://opensearch.org/docs/latest/search-plugins/vector-search/

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+To address this challenge, OpenSearch 2.17 introduced the concept of Application Based Configuration templates as an experimental feature. This feature allows users to easily configure their indices based on the specific use case they are building it for, reducing the need for manual tweaking and promoting a seamless on-boarding experience and managing the lifecycle of the index and its settings/mappings as new features are introduced.
```
Comment (pajuric) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:6]: Link the reference to OpenSearch 2.17 to the following: https://opensearch.org/blog/introducing-opensearch-2-17/

Comment (pajuric) [reply _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:6]: Lowercase application-based configuration  templates and link it to here: https://opensearch.org/docs/latest/im-plugin/index-context/

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+
+## What are Application-Based Configuration (ABC) Templates?
+
+Application-Based Configuration (ABC) templates in OpenSearch are predefined system templates designed to simplify the process of configuring indices for various use cases by providing predefined settings and configurations out of the box. These templates encapsulate various optimized settings, mappings, and configurations tailored for different use cases, eliminating the need for users to manually handle each setting or knob individually.
```
Comment (pajuric) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:11]: Sentence case - Application-based configuration templates for consistency.

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+---
+
+OpenSearch supports a wide variety of use cases---such as [logs](https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/logs/), [metrics](https://opensearch.org/docs/latest/monitoring-your-cluster/metrics/getting-started/), [traces](https://opensearch.org/docs/latest/data-prepper/common-use-cases/trace-analytics/), [website search](https://opensearch.org/docs/latest/search-plugins/), and [vectors](https://opensearch.org/docs/latest/search-plugins/vector-search/)---and enables you to build solutions for various applications based on your use case. As the use cases for OpenSearch continue to grow, managing indexes and configuring them with the right settings can become a daunting task, both for experienced and new users. OpenSearch provides numerous settings to fine-tune indexes for various performance and usability dimensions, such as throughput, latency, and disk utilization. However, for new users, finding the optimal configuration often requires extensive experimentation and developer effort, creating friction during the onboarding process. As new features are developed and released, experienced users of OpenSearch may overlook them as well.
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:14]:
```suggestion
OpenSearch supports a wide variety of use cases---such as [logs](https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/logs/), [metrics](https://opensearch.org/docs/latest/monitoring-your-cluster/metrics/getting-started/), [traces](https://opensearch.org/docs/latest/data-prepper/common-use-cases/trace-analytics/), [website search](https://opensearch.org/docs/latest/search-plugins/), and [vectors](https://opensearch.org/docs/latest/search-plugins/vector-search/)---and enables you to build solutions for various applications based on your use case. As the use cases for OpenSearch continue to grow, managing indexes and configuring them with the right settings can become a daunting task, both for experienced and new users. OpenSearch provides numerous settings that you can use to fine-tune indexes for various performance and usability dimensions, such as throughput, latency, and disk utilization. However, for new users, finding the optimal configuration often requires extensive experimentation and developer effort, creating friction during the onboarding process. As new features are developed and released, experienced OpenSearch users may overlook them as well.
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+To address this challenge, [OpenSearch 2.17](https://opensearch.org/blog/introducing-opensearch-2-17/) introduced the concept of _Application-Based Configuration (ABC) templates_ as an experimental feature. This feature allows you to easily configure your indexes based on your specific use case, reducing the need for manual updating while promoting a seamless onboarding experience and managing the lifecycle of the index and its settings and mappings as new features are introduced.
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:16]:
```suggestion
To address this challenge, [OpenSearch 2.17](https://opensearch.org/blog/introducing-opensearch-2-17/) introduced the concept of _Application-Based Configuration (ABC) templates_ as an experimental feature. This feature allows you to easily configure your indexes based on your specific use case, reducing the need for manual updates while promoting a seamless onboarding experience and managing the lifecycle of the index and its settings and mappings as new features are introduced.
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:17]: Should "Application-Based Configuration (ABC) templates" be used in the title?

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+ABC templates can be applied to your indexes or index templates by providing the `context` field in a request when creating an index or index template. This associates the index or index template with the specified use case (for example, logs or metrics), automatically applying the corresponding settings and mappings from the ABC template to the index or index template.
+
+ABC templates make the indexes use-case-aware and apply the best configuration applicable in terms of performance and end-to-end support at all stages, from index creation to data visualization (for example, Simple Schema for Observability [SS4O]). You can rely on composable index templates and component templates for providing the basic building blocks for the applicable configurations. ABC templates use component templates as a resource in order to expose use-case-specific configurations. The solution is designed to work well with existing index definitions, while the available optimizations are provided straight out of the box.
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:25]:
```suggestion
ABC templates make the indexes use-case-aware and apply the best applicable configuration in terms of performance and end-to-end support at all stages, from index creation to data visualization (for example, Simple Schema for Observability). You can rely on composable index templates and component templates for providing the basic building blocks for the applicable configurations. ABC templates use component templates as a resource in order to expose use-case-specific configurations. The solution is designed to work well with existing index definitions, while the available optimizations are provided straight out of the box.
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+
+## Why use ABC templates?
+
+Using ABC templates in OpenSearch offers the following advantages:
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:30]:
```suggestion
OpenSearch ABC templates offer the following advantages:
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+- **Simplified configuration**: ABC templates eliminate the need to navigate through numerous settings, reducing the complexity of index configuration.
+- **Optimized performance**: The predefined settings and mappings in the templates are optimized for specific use cases, ensuring better performance out of the box.
+- **Automatic updates**: As OpenSearch introduces new optimizations and features, they are added to the ABC template, and you can start using them for their newly created indexes with minimal effort.
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:34]:
```suggestion
- **Automatic updates**: As OpenSearch introduces new optimizations and features, they are added to the ABC templates, and you can start using them for newly created indexes with minimal effort.
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:35]: Line 34: I'm assuming that "their" shouldn't be here.

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+To understand the optimized settings by ABC templates and their performance benefits, we performed a comparison exercise using an index which relies on the `logs` template using HTTP logs dataset (at the same refresh interval), and found the following:
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:37]:
```suggestion
To understand the optimized ABC template settings and their performance benefits, we performed a comparison exercise using an index that relies on the `logs` template, which uses the HTTP logs dataset (at the same refresh interval), and found the following:
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:38]: Line 37: Confirm that my changes reflect the intended meaning.

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+- Storage improved by 20% (when using `zstd_no_dict` compared to the default `qat_lz4` compression for stored fields).
+- Indexing p99 latency improved by 6% because the generated segments are smaller and thus require fewer merges.
+- The template enforces that the index uses the `log_byte_size` merge policy, thus ensuring that the data ingested together is kept together even after segment merges, resulting in an improved performance of time range filter queries.
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:41]:
```suggestion
- The template requires the index to use the `log_byte_size` merge policy, thus ensuring that data ingested together is kept together even after segment merges, resulting in improved performance of time range filter queries.
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+The following diagram compares storage and p99 latency performance.
+
+![Application-based template performance comparison](/assets/media/blog-images/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates/perf-comparison.png)
+
+## How to use ABC templates?
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:47]:
```suggestion
## How to use ABC templates
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+To get started with ABC templates, see [Index context](https://opensearch.org/docs/latest/im-plugin/index-context/).
+
+The following animation illustrates creating an index using ABC templates. The template sets the required mappings and settings to the index. Additionally, it configures direct integration with OpenSearch dashboards for the data ingested thereafter.
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:51]:
```suggestion
The following animation illustrates creating an index using an ABC template. The template sets the required index mappings and settings. Additionally, it configures direct integration with OpenSearch Dashboards for subsequently ingested data.
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+### Limitations
+
+While providing a lot of benefits, ABC templates have the following limitations:
+
+* When using ABC templates for an index, you cannot include any settings defined in the template that supports the index context during its creation or during dynamic setting updates.
+* Once configured for an index, the context becomes permanent and cannot be removed.
+
+## Conclusion
+
+As OpenSearch continues to evolve, you can expect to benefit from new optimizations and features seamlessly integrated into your existing ABC templates, ensuring a consistent and optimized experience across various use cases. By providing predefined, optimized settings and configurations out of the box, this feature reduces the friction and developer effort required during the onboarding process. We look forward to your feedback about this feature on the [OpenSearch forum](https://forum.opensearch.org/), and working together to make the OpenSearch easier to use with every release.
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:77]:
```suggestion
As OpenSearch continues to evolve, you can expect to benefit from new optimizations and features seamlessly integrated into your existing ABC templates, ensuring a consistent and optimized experience across various use cases. By providing predefined, optimized settings and configurations out of the box, this feature reduces the friction and developer effort required during the onboarding process. We look forward to your feedback on the [OpenSearch forum](https://forum.opensearch.org/) and to working together to make OpenSearch easier to use with every version release. 
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+The following diagram compares storage and p99 latency performance.
+
+![Application-based template performance comparison](/assets/media/blog-images/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates/perf-comparison.png)
+
+## How to use ABC templates?
+
+To get started with ABC templates, see [Index context](https://opensearch.org/docs/latest/im-plugin/index-context/).
+
+The following animation illustrates creating an index using ABC templates. The template sets the required mappings and settings to the index. Additionally, it configures direct integration with OpenSearch dashboards for the data ingested thereafter.
+
```
Comment (natebower) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:52]: Line 51: Confirm that my change to "an ABC template" is correct (following "The template" in the next sentence).

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @mgodwan Editorial review complete. Nicely written 😄. Please see my comments and changes and let me know if you have any questions. Thanks!

Cc: @pajuric

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 18]
```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title: "OpenSearch simplified: The power of application-based configuration templates"
```
Comment (kolchfa-aws) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:3]:
```suggestion
title: "OpenSearch simplified: The power of Application-Based Configuration (ABC) templates"
```

**Comment by kolchfa-aws** [comment]
@pajuric Could you provide the meta for this blog?

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+title: "OpenSearch simplified: The power of Application-Based Configuration (ABC) templates"
+authors:
+   - mgodwani
+   - kolchfa
+date: 2024-12-19
+categories: 
+    - technical-post
+meta_keywords: 
+meta_description: 
+---
```
Comment (pajuric) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:12]: Please add the following meta: 

_meta_keywords:_ Application-Based Configuration templates, ABC templates, index templates, index configuration, OpenSearch performance optimization

_meta_description:_ Explore OpenSearch's Application-Based Configuration (ABC) templates simplifies index configuration, optimizes performance, and streamlines setup for various use cases including logs, metrics, and traces.

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+layout: post
+title: "OpenSearch simplified: The power of Application-Based Configuration (ABC) templates"
+authors:
+   - mgodwani
+   - kolchfa
+date: 2024-12-19
+categories: 
+    - technical-post
+meta_keywords:
```
Comment (kolchfa-aws) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:10]:
```suggestion
meta_keywords: Application-Based Configuration templates, ABC templates, index templates, index configuration, OpenSearch performance optimization
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+layout: post
+title: "OpenSearch simplified: The power of Application-Based Configuration (ABC) templates"
+authors:
+   - mgodwani
+   - kolchfa
+date: 2024-12-19
+categories: 
+    - technical-post
+meta_keywords: Application-Based Configuration templates, ABC templates, index templates, index configuration, OpenSearch performance optimization
+meta_description:
```
Comment (kolchfa-aws) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:11]:
```suggestion
meta_description: Explore OpenSearch's Application-Based Configuration (ABC) templates simplifies index configuration, optimizes performance, and streamlines setup for various use cases including logs, metrics, and traces.
```

File: _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown [hunk 22]
```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title: "OpenSearch simplified: The power of Application-Based Configuration (ABC) templates"
+authors:
+   - mgodwani
+   - kolchfa
+date: 2024-12-19
```
Comment (kolchfa-aws) [inline _posts/2024-12-20-OpenSearch-Simplified-The-Power-of-Application-Based-Templates.markdown:7]:
```suggestion
date: 2024-12-23
```

**Comment by pajuric** [comment]
@nateynateynate @krisfreedain - Blog is ready to publish today. Thanks all and Happy Holidays!

**Comment by pajuric** [comment]
@nateynateynate @krisfreedain - Let's get this one published on 12/26, please.

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**